core/src/main/java/org/apache/iceberg/CachingCatalog.java

```diff
@@ -197,6 +197,14 @@ public Table registerTable(TableIdentifier identifier, String metadataFileLocati
     return table;
   }
 
+  @Override
+  public Table registerTable(
```

Comment:
I don't think you need to keep the other Override for registerTable() within this class. Catalog.registerTable(no override flag) would take care of it anyway.